### ISSUE TITLE
[ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:

### ISSUE THREAD

**Issue by bravekingzhang** [renefloor/flutter_cached_network_image#35]
```
E/flutter (18565): [ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:
E/flutter (18565): 'package:flutter/src/painting/image_stream.dart': Failed assertion: line 362 pos 12: '_codec != null': is not true.
E/flutter (18565): #0      _AssertionError._doThrowNew (dart:core/runtime/liberrors_patch.dart:37:39)
E/flutter (18565): #1      _AssertionError._throwNew (dart:core/runtime/liberrors_patch.dart:33:5)
E/flutter (18565): #2      MultiFrameImageStreamCompleter._handleCodecReady (package:flutter/src/painting/image_stream.dart)
E/flutter (18565): #3      _RootZone.runUnary (dart:async/zone.dart:1381:54)
E/flutter (18565): #4      _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
E/flutter (18565): #5      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:637:45)
E/flutter (18565): #6      Future._propagateToListeners (dart:async/future_impl.dart:666:32)
E/flutter (18565): #7      Future._completeError (dart:async/future_impl.dart:489:5)
E/flutter (18565): #8      _SyncCompleter._completeError (dart:async/future_impl.dart:55:12)
E/flutter (18565): #9      _Completer.completeError (dart:async/future_impl.dart:27:5)
E/flutter (18565): #10     CachedNetworkImageProvider._loadAsync (package:cached_network_image/cached_network_image.dart)
E/flutter (18565): <asynchronous suspension>
E/flutter (18565): #11     CachedNetworkImageProvider.load (package:cached_network_image/cached_network_image.dart:462:16)
E/flutter (18565): #12     ImageProvider.resolve.<anonymous closure>.<anonymous closure> (package:flutter/src/painting/image_provider.dart:265:86)
E/flutter (18565): #13     ImageCache.putIfAbsent (package:flutter/src/painting/image_cache.dart:82:22)
E/flutter (18565): #14     ImageProvider.resolve.<anonymous closure> (package:flutter/src/painting/image_provider.dart:265:63)
E/flutter (18565): #15     SynchronousFuture.then (package:flutter/src/foundation/synchronous_future.dart:38:29)
E/flutter (18565): #16     ImageProvider.resolve (package:flutter/src/painting/image_provider.dart:263:30)
E/flutter (18565): #17     _ImageProviderResolver.resolve (package:cached_network_image/cached_network_image.dart:193:29)
E/flutter (18565): #18     _CachedNetworkImageState._resolveImage (package:cached_network_image/cached_network_image.dart:284:20)
E/flutter (18565): #19     _CachedNetworkImageState.didChangeDependencies (package:cached_network_image/cached_network_image.dart:261:5)
E/flutter (18565): #20     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3750:12)
E/flutter (18565): #21     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600:5)
E/flutter (18565): #22     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
E/flutter (18565): #23     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
E/flutter (18565): #24     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4637:14)
E/flutter (18565): #25     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
E/flutter (18565): #26     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
E/flutter (18565): #27     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636:16)
E/flutter (18565): #28     Element.rebuild (package:flutter/src/widgets/framework.dart:3478:5)
E/flutter (18565): #29     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3605:5)
E/flutter (18565): #30     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600:5)
E/flutter (18565): #31     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
E/flutter (18565): #32     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4742:32)
E/flutter (18565): #33     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
E/flutter (18565): #34     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4742:32)
E/flutter (18565): #35     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
E/flutter (18565): #36     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
E/flutter (18565): #37     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4637:14)
E/flutter (18565): #38     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
E/flutter (18565): #39     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
E/flutter (18565): #40     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636:16)
E/flutter (18565): #41     Element.rebuild (package:flutter/src/widgets/framework.dart:3478:5)
E/flutter (18565): #42     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3605:5)
E/flutter (18565): #43     ComponentElement.mount (package:flut
```

**Comment by bravekingzhang** [comment]
0.4.0
this bug can be re found like this.

first ,you show image use Image.network....

and then change all the  image widget into cacheImage,.


**how to fix it:**

uninstall your app. and launch again ,things will be right now

**Comment by renefloor** [comment]
It looks like there is something going wrong with the memory image_cache of flutter itself. When would you want to combine Image.network and CachedNetworkImage?

**Comment by renefloor** [comment]
Can you check if you problems are solved with version 0.4.1?

**Comment by virenvvarasadiya** [comment]
Remaining Errors !!!
Any Solution ??
 
#0      IOClient.send (package:http/src/io_client.dart:30:23)
    <asynchronous suspension>
    #1      BaseClient._sendUnstreamed (package:http/src/base_client.dart:171:38)
    <asynchronous suspension>
    #2      BaseClient.get (package:http/src/base_client.dart:34:5)
    #3      get.<anonymous closure> (package:http/http.dart:47:34)
    #4      _withClient (package:http/http.dart:167:20)
    <asynchronous suspension>
    #5      get (package:http/http.dart:47:3)
    #6      _MyHomePageState.getData (file:///F:/flutter/database/lib/main.dart:28:41)
    <asynchronous suspension>
    #7      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:494:14)
    #8      _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:549:30)
    #9      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:102:24)
    #10     TapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:161:9)
    #11     TapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:94:7)
    #12     PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:315:9)
    #13     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:73:12)
    #14     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:101:11)
    #15     _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:143:19)
    #16     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:121:22)
    #17     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:101:7)
    #18     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:64:7)
    #19     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:48:7)
    #20     _invoke1 (dart:ui/hooks.dart:134:13)
    #21     _dispatchPointerDataPacket (dart:ui/hooks.dart:91:5)
